link/raw_tracepoint.go

```diff
@@ -33,29 +34,57 @@ func AttachRawTracepoint(opts RawTracepointOptions) (Link, error) {
 		return nil, err
 	}
 
-	return &progAttachRawTracepoint{fd: fd}, nil
+	if err := haveBPFLink(); errors.Is(err, ErrNotSupported) {
+		// Prior to commit 70ed506c3bbc ("bpf: Introduce pinnable bpf_link abstraction")
+		// raw_tracepoints are just a plain fd.
+		return &simpleRawTracepoint{fd}, nil
+	} else if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	if err = haveBPFLink(); errors.Is(err, ErrNotSupported) {
		// Prior to commit 70ed506c3bbc ("bpf: Introduce pinnable bpf_link abstraction")
		// raw_tracepoints are just a plain fd.
		return &simpleRawTracepoint{fd}, nil
	}
	if err != nil {
		return nil, err
	}
```

Reply:
It took me a bit to figure out what your suggestion is about, basically you don't want to shadow `err`?

Reply:
I just wanted to remove extra `else` once we have return at first if statement. a bit cleaner.

Reply:
Got rid of the else.